core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/DefaultEncodingFactory.java

```diff
@@ -103,6 +103,7 @@ private ColumnPageEncoder createEncoderForDimensionLegacy(TableSpec.DimensionSpe
         return new HighCardDictDimensionIndexCodec(
             dimensionSpec.isInSortColumns(),
```

Comment:
Instead of passing all the properties of dimensionspec individually, pass dimensionspec it self.

Reply:
emm, better not do this in this PR. All the parameters for *IndexCodec looks alike. Changing all of them will introduce unrelated changes.